eli5/formatters/html.py

```diff
@@ -119,12 +120,14 @@ def render_targets_weighted_spans(targets, preserve_density):
 
 def render_weighted_spans(pws):
     # type: (PreparedWeightedSpans) -> str
-    # TODO - can be much smarter, join spans at least
-    # TODO - for longer documents, remove text without active features
+    # TODO - for longer documents, an remove text without active features
     return ''.join(
-        _colorize(token, weight, pws.weight_range)
-        for token, weight in zip(
-            pws.doc_weighted_spans.document, pws.char_weights))
+        _colorize(''.join(t for t, _ in tokens_weights),
+                  weight,
+                  pws.weight_range)
+        for weight, tokens_weights in groupby(
+            zip(pws.doc_weighted_spans.document, pws.char_weights),
```

Comment:
Even if formatted weight precision is higher than noticeable in colors, we still have to show weights in title, so we can't join spans with different formatted weights.

Color range can't be more than 256 (as we're using a single HSL component), and it seems for us it is even less because the range is limited.